katello/katello-change-hostname

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/ruby
+
+require "socket"
+require "optparse"
+
+DEFAULT_PROGRAM = "foreman"
+DEFAULT_SCENARIO = "katello"
+DEFAULT_SSL_DIR = "/root/ssl-build"
+
+def is_rhel_version?(version)
+  `cat /etc/redhat-release`[/(\d+\.)/].include?(version)
+end
+
+def timestamp
+  Time.now.strftime("%Y%m%d%H%M")
+end
+
+def get_hostname
+  if is_rhel_version?("6")
+    hostname = `hostname`
+  else
+    hostname = Socket.gethostname
+  end
+  hostname.chomp 
+end
+
+options = {}
+options[:program] = DEFAULT_PROGRAM
+options[:scenario] = DEFAULT_SCENARIO
+options[:ssl_dir] = DEFAULT_SSL_DIR
+
+opt_parser = OptionParser.new do |opt|
+  opt.banner = "Usage: katello-change-hostname HOSTNAME [OPTIONS]"
+  opt.separator  ""
+  opt.separator  "Example:"
+  opt.separator  " katello-change-hostname foo.example.com -u admin -p changeme"
+  opt.separator  ""
+  opt.separator  "Options"
+
+  opt.on("-u","--username USERNAME","admin username REQUIRED") do |username|
+    options[:username] = username
+  end
+
+  opt.on("-p","--password PASSWORD","admin password REQUIRED") do |password|
+    options[:password] = password
+  end
+
+  opt.on("-g","--program PROGRAM","name of the program you are modifying (defaults to #{DEFAULT_PROGRAM})") do |program|
+    options[:program] = program
+  end
+
+  opt.on("-s","--scenario SCENARIO","name of the scenario you are modifying (defaults to #{DEFAULT_SCENARIO})") do |scenario|
+    options[:scenario] = scenario
+  end
+
+  opt.on("-d","--disable-system-checks","runs the installer with --disable-system-checks") do |system_check|
+    options[:system_check] = true
+  end
+
+  opt.on("-l","--ssl-dir SSL_DIR","specify the ssl cert directory to replace (defaults to #{DEFAULT_SSL_DIR})") do |ssl_dir|
+    options[:ssl_dir] = ssl_dir
+  end
+
+  opt.on("-h","--help","help") do
+    puts opt_parser
+    exit
+  end
+end
+opt_parser.parse!
+
+raise 'Must run as root' unless Process.uid == 0
+
+unless options[:username] && options[:password]
+  STDOUT.puts "Username and/or Password options are missing!"
+  puts opt_parser
+  exit
+end
+
+if options[:scenario] == "katello"
+  proxy = "Smart Proxy"
+  plural_proxy = "Smart Proxies"
+else
+  proxy = "Capsule"
+  plural_proxy = "Capsules"
+end
+
+if ARGV[0] && ARGV.count >= 1
+  new_hostname = ARGV[0]
+else
+  puts opt_parser
+  exit
+end
+
+# Get the hostname from your system
+old_hostname = get_hostname
+
+STDOUT.puts "Updating default #{proxy}"
+default_capsule_id = `hammer -u #{options[:username]} -p #{options[:password]} capsule list | grep #{old_hostname} | awk '{ print $1 }'`[0]
+capsule_hammer = `hammer -u #{options[:username]} -p #{options[:password]} capsule update --id #{default_capsule_id} --url https://#{new_hostname}:9090 --new-name #{new_hostname}`
```

Comment:
Looks like its trying to contact the new url. The funny thing is I can update these values in the UI just fine

```
[vagrant@centos7-katello-3-2 ~]$ hammer -u admin -p changeme -d capsule update --id 1 --url https://foo.example.com:9090 --new-name foo.example.com
[ INFO 2017-01-05 21:47:19 Init] Initialization of Hammer CLI (0.8.0) has started...
[DEBUG 2017-01-05 21:47:19 Init] Running at ruby 2.2.2-p95
[ INFO 2017-01-05 21:47:19 Init] Configuration from the file /etc/hammer/cli_config.yml has been loaded
[ INFO 2017-01-05 21:47:19 Init] Configuration from the file /etc/hammer/cli.modules.d/foreman.yml has been loaded
[ INFO 2017-01-05 21:47:19 Init] Configuration from the file /etc/hammer/cli.modules.d/foreman_bootdisk.yml has been loaded
[ INFO 2017-01-05 21:47:19 Init] Configuration from the file /etc/hammer/cli.modules.d/foreman_docker.yml has been loaded
[ INFO 2017-01-05 21:47:19 Init] Configuration from the file /etc/hammer/cli.modules.d/foreman_tasks.yml has been loaded
[ INFO 2017-01-05 21:47:19 Init] Configuration from the file /etc/hammer/cli.modules.d/import.yml has been loaded
[ INFO 2017-01-05 21:47:19 Init] Configuration from the file /etc/hammer/cli.modules.d/katello.yml has been loaded
[DEBUG 2017-01-05 21:47:19 Connection] Registered: foreman
[DEBUG 2017-01-05 21:47:19 API] Global headers: {
        :content_type => "application/json",
              :accept => "application/json;version=2",
    "Accept-Language" => "en"
}
[DEBUG 2017-01-05 21:47:19 API] Follow redirects: never
[ INFO 2017-01-05 21:47:20 Modules] Extension module hammer_cli_foreman (0.8.0) loaded
[ INFO 2017-01-05 21:47:20 Modules] Extension module hammer_cli_foreman_bootdisk (0.1.3) loaded
[ INFO 2017-01-05 21:47:20 Modules] Extension module hammer_cli_import (0.11.3) loaded
[ INFO 2017-01-05 21:47:20 HammerCLI::MainCommand] subcommand organization (HammerCLIForeman::Organization) was removed.
[ INFO 2017-01-05 21:47:20 HammerCLI::MainCommand] subcommand organization (HammerCLIKatello::Organization) was created.
[ INFO 2017-01-05 21:47:20 Modules] Extension module hammer_cli_katello (0.1.3) loaded
[DEBUG 2017-01-05 21:47:20 Init] Using locale 'en'
[DEBUG 2017-01-05 21:47:20 Init] 'mo' files for locale domain 'hammer-cli' loaded from '/opt/theforeman/tfm/root/usr/share/gems/gems/hammer_cli-0.8.0/locale'
[DEBUG 2017-01-05 21:47:20 Init] 'mo' files for locale domain 'hammer-cli-foreman' loaded from '/opt/theforeman/tfm/root/usr/share/gems/gems/hammer_cli_foreman-0.8.0/locale'
[DEBUG 2017-01-05 21:47:20 Init] 'mo' files for locale domain 'hammer_cli_foreman_docker' loaded from '/opt/theforeman/tfm/root/usr/share/gems/gems/hammer_cli_foreman_docker-0.0.4/locale'
[DEBUG 2017-01-05 21:47:20 Init] 'mo' files for locale domain 'hammer-cli-katello' loaded from '/opt/theforeman/tfm/root/usr/share/gems/gems/hammer_cli_katello-0.1.3/locale'
[ INFO 2017-01-05 21:47:20 HammerCLI::MainCommand] Called with options: {"option_debug"=>true, "option_username"=>"admin", "option_password"=>"***"}
[ INFO 2017-01-05 21:47:20 HammerCLIForeman::SmartProxy] Called with options: {}
[ INFO 2017-01-05 21:47:20 HammerCLIForeman::SmartProxy::UpdateCommand] Called with options: {"option_id"=>"1", "option_url"=>"https://foo.example.com:9090", "option_new_name"=>"foo.example.com"}
[ INFO 2017-01-05 21:47:20 API] Server: https://localhost/
[ INFO 2017-01-05 21:47:20 API] PUT /api/smart_proxies/1
[DEBUG 2017-01-05 21:47:20 API] Params: {
    "smart_proxy" => {
         "url" => "https://foo.example.com:9090",
        "name" => "foo.example.com"
    },
           "name" => "foo.example.com"
}
[DEBUG 2017-01-05 21:47:20 API] Headers: {}
[ERROR 2017-01-05 21:47:23 API] 422 Unprocessable Entity
[DEBUG 2017-01-05 21:47:23 API] {
    "errors" => {
        "base" => [
            [0] "Unable to communicate with the proxy: ERF12-2530 [ProxyAPI::ProxyException]: Unable to detect features ([Errno::EHOSTUNREACH]: No route to host - connect(2) for \"foo.example.com\" port 9090) for proxy https://foo.example.com:9090/features",
            [1] "Please check the proxy is configured and running on the host."
        ]
    }
}
[ERROR 2017-01-05 21:47:23 Exception] <NilClass> nil
Could not update the proxy
```

```
[vagrant@centos7-katello-3-2 ~]$ rpm -qa | grep hammer
tfm-rubygem-hammer_cli_katello-0.1.3-1.el7.noarch
tfm-rubygem-hammer_cli-0.8.0-1.el7.noarch
tfm-rubygem-hammer_cli_foreman_bootdisk-0.1.3-5.el7.noarch
tfm-rubygem-hammer_cli_foreman_tasks-0.0.10-2.fm1_12.el7.noarch
tfm-rubygem-hammer_cli_import-0.11.3-1.el7.noarch
tfm-rubygem-hammer_cli_foreman-0.8.0-1.el7.noarch
tfm-rubygem-hammer_cli_foreman_docker-0.0.4-2.el7.noarch
```